milatools/datasets/cached.py

```diff
@@ -0,0 +1,48 @@
+import os
+import shutil
+
+
+class CachedDataset:
+    """Simple dataset that moves the dataset from a shared network location to a fast local location"""
+
+    WAS_COPIED = False
+    train_size = None
+    valid_size = None
+    test_size = None
+
+    def __init__(self, root, download=False):
+        if os.path.exists(self.mila_path()):
+            download = False
+
+        if not CachedDataset.WAS_COPIED:
+            shutil.copytree(self.mila_path(), root, dirs_exist_ok=True)
+            CachedDataset.WAS_COPIED = True
```

Comment:
We can use [Value][1] for multiprocessing. 
Yeah, it might be an issue for multiple datasets

[1]: https://www.kite.com/python/docs/multiprocessing.Value

Reply:
Actually with the distributed module we can use has_dataset_authority() to know if the worker should be downloading it or not. That is how pytorch usually does it

Reply:
So `WAS_COPIED` is going to be fine, it is not shared.

```
python
Python 3.8.3 (default, May 19 2020, 18:47:26) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> class A:
...     i = 2
... 
>>> class B(A):
...     pass
... 
>>> A.i
2
>>> B.i
2
>>> B.i = 3
>>> A.i
2
```

Reply:
My point is that we're not setting the attribute on the subclasses, the code always reads or writes the attribute on `CachedDataset`. To have it work on a per class basis we would need to use `type(self).WAS_COPIED`.

Reply:
I did the modification